nss/integration-tests/nss_test.go

```diff
@@ -0,0 +1,176 @@
+package nss_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/testutils"
+)
+
+var libPath string
+var rustCovEnv []string
+
+func TestIntegration(t *testing.T) {
+	t.Parallel()
+
+	buildRustNSSLib(t)
+
+	originOuts := map[string]string{}
+	for _, db := range []string{"passwd", "group", "shadow"} {
+		//#nosec:G204 - We control the cmd arguments in tests.
+		data, err := exec.Command("getent", db).CombinedOutput()
+		require.NoError(t, err, "Setup: can't run getent to get original output from system")
+		originOuts[db] = string(data)
+	}
+
+	tests := map[string]struct {
+		db      string
+		key     string
+		cacheDB string
+
+		noDaemon       bool
+		noCustomSocket bool
+
+		wantErr bool
+	}{
+		"Get all entries from passwd": {db: "passwd"},
+		"Get all entries from group":  {db: "group"},
+		"Get all entries from shadow": {db: "shadow"},
+
+		"Get entry from passwd by name": {db: "passwd", key: "user1"},
+		"Get entry from group by name":  {db: "group", key: "group1"},
+		"Get entry from shadow by name": {db: "shadow", key: "user1"},
+
+		"Get entry from passwd by id": {db: "passwd", key: "1111"},
+		"Get entry from group by id":  {db: "group", key: "11111"},
+
+		/* Error cases */
+		// Even though those are "error" cases, the getent command won't fail since the other databases on the machine will return some entries.
+		"Uses default socket if env is not set":                                          {db: "passwd", noCustomSocket: true},
```

Comment:
I find the test misleading: it just tells that it uses default socket if env is not set, but returns nothing.

Reply:
It should return nothing... The daemon is not running on the default socket.

Reply:
the test case name does not match the content for sure.

However, there is no more default socket path on the daemon side, only on the pam/nss side, no? I’m afraid that we hit the production service and the tests will fails on some configuration.

Reply:
But this is on the NSS side... The socket is configured on the daemon, but the NSS module connects either to `/run/authd.sock` or to the value of `$AUTHD_NSS_SOCKET`.
It's definitely a miss on our side. We updated the daemon to have a configurable socket path, but we didn't think about how to handle this on the NSS side. Any ideas on how should we handle this?